preinit_connection.go

```diff
@@ -93,7 +98,12 @@ func (ch *Channel) inboundHandshake(ctx context.Context, c net.Conn, events conn
 		return nil, err
 	}
 
-	return ch.newConnection(c, 0 /* initialID */, "" /* outboundHP */, remotePeer, remotePeerAddress, events), nil
+	baseCtx := context.Background()
+	if p := getTChannelParams(ctx); p != nil {
```

Comment:
the only caller of this function is in `channel.go` which passes `context.Background()`, so this can never happen. should we delete this (to make it more explicit, we can remove the context argument from this function and just pass `context.Background()` below directly.